src/python/foglamp/core/server.py

```diff
@@ -30,6 +51,300 @@ class Server:
     scheduler = None
     """ foglamp.core.Scheduler """
 
+    logging_configured = False
+    """Set to true when it's safe to use logging"""
+
+    # TODO: Discuss if this needs to be pre-decided and fixed
+    _MANAGEMENT_API_PORT = 8081
+
+    # TODO: Set below storage ports to None after Storage layer is fixed to accept these values via command line arguments
+    _STORAGE_PORT = 8080
+    _STORAGE_MANAGEMENT_PORT = 1081
+
+    # TODO: Fix a TODO in _start() below to accept below via discovery
+    _RESTAPI_PORT = 8082
+
+    @staticmethod
+    def request_available_port():
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.bind(('localhost', 0))
+        addr, port = s.getsockname()
+        s.close()
+        return port
+
+    @staticmethod
+    def _safe_make_dirs(path):
+        """Creates any missing parent directories
+
+        :param path: The path of the directory to create
+        """
+
+        try:
+            os.makedirs(path, 0o750)
+        except OSError as exception:
+            if not os.path.exists(path):
+                raise exception
+
+    @classmethod
+    def _configure_logging(cls):
+        """Alters the root logger to send messages to syslog
+           with a filter of WARNING
+        """
+        if cls.logging_configured:
+            return
+
+        logger.setup()
+        cls.logging_configured = True
+
+
+
+    """ Management API """
+    @staticmethod
+    def _make_management():
+        """Creates the REST server
+
+        :rtype: web.Application
+        """
+        core = web.Application(middlewares=[middleware.error_middleware])
+        routes_core.setup(core)
+        return core
+
+    @classmethod
+    def _run_management_api(cls, _MANAGEMENT_PORT):
+        web.run_app(cls._make_management(), host='0.0.0.0', port=_MANAGEMENT_PORT)
+
+    @staticmethod
+    def _get_management_pid():
+        """Returns FogLAMP's process id or None if FogLAMP is not running"""
+
+        try:
+            with open(_MANAGEMENT_PID_PATH, 'r') as pid_file:
+                pid = int(pid_file.read().strip())
+        except (IOError, ValueError):
+            return None
+
+        # Delete the pid file if the process isn't alive
+        # there is an unavoidable race condition here if another
+        # process is stopping or starting the daemon
+        try:
+            os.kill(pid, 0)
+        except OSError:
+            os.remove(_MANAGEMENT_PID_PATH)
+            pid = None
+
+        return pid
+
+    @classmethod
+    def _start_management_service(cls):
+        # Start Management API
+        print("Starting Management API")
+        try:
+            cls._safe_make_dirs(os.path.dirname(_MANAGEMENT_PID_PATH))
+            setproctitle.setproctitle('management')
```

Comment:
What is this for?

Reply:
@MarkRiddoch to rename a process... for example: it is also being used to rename core process to foglamp